packages/back-end/src/models/CustomFieldModel.ts

```diff
@@ -26,15 +26,33 @@ export class CustomFieldModel extends BaseClass {
   }
 
   protected canCreate(): boolean {
-    return this.context.permissions.canManageCustomFields();
+    if (!this.context.hasPremiumFeature("custom-metadata")) {
+      throw new Error(
+        "Your organization's plan does not include the custom fields feature.",
+      );
+    }
+
+    if (!this.context.permissions.canManageCustomFields()) {
+      throw this.context.permissions.throwPermissionError();
+    }
```

Comment:
I don't think this is right. The can* methods should return a Boolean and the base class should do the throwing. If the status code being returned is wrong, we should fix that in the base class.

Reply:
Also, we should probably add a new method to the base class `hasPremiumFeature` instead that returns true by default (but can be overriden). We can check that automatically during create and update operations and throw if needed.

Reply:
I see.

Is the idea of the `can*` methods that those only check permissions? If so, that makes the update to the BaseModel class pretty simple - we can update that to return a permission error, instead of the generic error.

If not, then it'll take a bit more work as the `can*` methods could return false for a number of reasons. 

This is tangental, but if the `can*` methods are only permission checks, I'm not sure the naming has enough context.

I can create the new `hasPremiuimFeature` method to the base class and check that before doing the create & update operations.